crypto/crypto.go

```diff
@@ -215,7 +230,27 @@ func NewNaiveSigner(privateKey PrivateKey, hashAlgo HashAlgorithm) NaiveSigner {
 
 // GeneratePrivateKey generates a private key with the specified signature algorithm from the given seed.
 func GeneratePrivateKey(sigAlgo SignatureAlgorithm, seed []byte) (PrivateKey, error) {
-	privKey, err := crypto.GeneratePrivateKey(crypto.SigningAlgorithm(sigAlgo), seed)
+	// check the seed has minimum entropy
+	if len(seed) < MinSeedLength {
+		return PrivateKey{}, fmt.Errorf(
+			"crypto: insufficient seed length %d, must be at least %d bytes for %s",
+			len(seed),
+			MinSeedLength,
+			sigAlgo,
+		)
+	}
+
+	// expand the seed and uniformize its entropy
+	generationTag := []byte("ECDSA Key Generation")
+	customizer := []byte("")
+	hasher, err := hash.NewKMAC_128(generationTag, customizer, sigAlgo.minSeedLength())
```

Comment:
Maybe we could remove `sigAlgo.minSeedLength()` and instead put this inline:

```go
        var seedLen int
	switch f {
	case ECDSA_P256:
		seedLen = crypto.KeyGenSeedMinLenECDSAP256
	case ECDSA_secp256k1:
		seedLen = crypto.KeyGenSeedMinLenECDSASecp256k1
        default:
                 return PrivateKey{}, fmt.Errorf("crypto: invalid signature algorithm %s", sigAlgo)
	}
```

Reply:
Yeah I agree